### PR TITLE
Redirect sso on oauth

### PR DESCRIPTION
**Pull request by kahlouie**
This redirects to the sso url when you've been logged out.

### CODE REVIEW

**Comment by naaman** [comment]
@kahlouie Are we sure this is the right thing to do? `/logout` is an explicit logout, not session expiration. I think we're going to end up removing this when we implement the [single logout](https://www.pivotaltracker.com/story/show/101931460).

**Comment by kahlouie** [comment]
@dmcinnes We decided this is the right path for the moment. I'm happy to talk about this a bit more one on one.

**Comment by geemus** [comment]
oops, misread that as decision against :(

**Comment by kahlouie** [comment]
@geemus no worries!

File: lib/identity/auth.rb [hunk 1]
```diff
@@ -332,13 +332,15 @@ def get_client_campaign(oauth_client_id)
     end
 
     def logout
-      @cookie.clear
-
-      url = if params[:url] && safe_redirect?(params[:url])
+      url = if !!@cookie.sso_entity &&  Config.sso_base_url
+        "#{Config.sso_base_url}/#{@cookie.sso_entity}"
```
Comment (dmcinnes) [inline lib/identity/auth.rb]: Can you extract this logic/construction into another method?
Also this:
https://github.com/heroku/identity/blob/redirect-sso-on-oauth/lib/identity/auth.rb#L22-L24

**Comment by dmcinnes** [comment]
LGTM, just that one comment!

**Comment by dmcinnes** [comment]
Thanks!

**Pull request merged**